cranelift/fuzzgen/src/function_generator.rs

```diff
@@ -454,15 +460,15 @@ fn valid_for_target(triple: &Triple, op: Opcode, args: &[Type], rets: &[Type]) -
             assert_eq!(rets.len(), 1);
 
             let arg = args[0];
-            let ret = args[0];
+            let ret = rets[0];
 
             // Vector arguments must produce vector results, and scalar arguments must produce
             // scalar results.
             if arg.is_vector() != ret.is_vector() {
                 return false;
             }
 
-            if arg.is_vector() && arg.is_vector() {
+            if arg.is_vector() && ret.is_vector() {
```

Comment:
This is a little bit of a driveby fix, but these looked wrong. I ran fuzzgen afterwards and it didn't reveal any new panics.